src/InstrumentationEngine/Instruction.h

```diff
@@ -159,25 +163,25 @@ namespace MicrosoftInstrumentationEngine
         bool GetIsBranchInternal() const { return IsFlagSet(s_ilOpcodeInfo[m_opcode].m_flags, ILOpcodeFlag_Branch); }
 
     protected:
-        template<typename... Args>
-        HRESULT QueryInterfaceInternal(_In_ REFIID riid, _Out_ void** ppvObject, Args... pThisArgs)
+
+        template<typename TBase>
+        HRESULT QueryBaseType(_In_ REFIID riid, _Out_ void** ppvObject)
```

Comment:
Would it be easier if we folded QueryBaseType and QueryInterfaceInternal into one function? Something like:
```c++
template<typename TBase, typename... Args>
QueryInterfaceInternal(...)
```
````